File: Leetcode/Implement strStr().py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        l, r = 0, len(needle)
        
        while r < len(haystack) + 1:
            if haystack[l:r] == needle:
                return l
            l += 1
            r += 1
            
        return -1

File: Leetcode/Implement strStr().java
class Solution {
    public int strStr(String haystack, String needle) {
        int left = 0;
        int right = needle.length();
        while (right <= haystack.length()){
            if(haystack.substring(left, right).equals(needle)){
                return left;
            }
            left += 1;
            right += 1;
        }
        return -1;
    }
}
